src/__tests__/fixtures/regression.css

```diff
@@ -0,0 +1,7 @@
+@custom-selector :--button .button;
+
+:--button {
+  &--active {
```

Comment:
This issue https://github.com/jonathantneal/postcss-nesting/issues/17 confirm that this behavior should not work.